gluten-substrait/src/main/scala/org/apache/gluten/execution/WholeStageTransformer.scala

```diff
@@ -298,27 +298,13 @@ case class WholeStageTransformer(child: SparkPlan, materializeInput: Boolean = f
       wsCtx: WholeStageTransformContext,
       inputRDDs: ColumnarInputRDDsWrapper,
       pipelineTime: SQLMetric): RDD[ColumnarBatch] = {
-    val isKeyGroupPartition = leafTransformers.exists {
-      // TODO: May can apply to BatchScanExecTransformer without key group partitioning
-      case b: BatchScanExecTransformerBase if b.keyGroupedPartitioning.isDefined => true
-      case _ => false
-    }
 
     /**
      * If containing leaf exec transformer this "whole stage" generates a RDD which itself takes
      * care of [[LeafTransformSupport]] there won't be any other RDD for leaf operator. As a result,
      * genFirstStageIterator rather than genFinalStageIterator will be invoked
      */
-    val allInputPartitions = leafTransformers.map(
-      leafTransformer => {
-        if (isKeyGroupPartition) {
-          leafTransformer.asInstanceOf[BatchScanExecTransformerBase].getPartitionsWithIndex
-        } else {
-          Seq(leafTransformer.getPartitions)
-        }
-      })
-
-    val allSplitInfos = getSplitInfosFromPartitions(isKeyGroupPartition, leafTransformers)
+    val allSplitInfos = leafTransformers.map(_.getSplitInfos).transpose
```

Comment:
Do we need to preserve a similar comment like [this](https://github.com/apache/incubator-gluten/blob/84f040f5fd57e3940bfba17713dd55ba47287224/gluten-substrait/src/main/scala/org/apache/gluten/execution/WholeStageTransformer.scala#L373-L408)? Which will help user understand the `.transpose` usage here.